examples/mongodbatlas_advanced_cluster/effective_fields/existing_module/README.md

```diff
@@ -0,0 +1,253 @@
+
+The migration enables module users to upgrade by simply changing the module source, with no other configuration changes required.
+
+## Directory Structure
+
+```
+existing_module/
+├── README.md                  # This file
+├── module_v1/                 # v1 module with lifecycle.ignore_changes
+│   ├── main.tf
+│   ├── variables.tf
+│   ├── outputs.tf
+│   └── versions.tf
+├── module_v2/                 # Modernized module with use_effective_fields
+│   ├── main.tf
+│   ├── variables.tf
+│   ├── outputs.tf
+│   └── versions.tf
+├── usage_v1/                  # Example using module_v1
+│   ├── main.tf
+│   ├── variables.tf
+│   └── versions.tf
+└── usage_v2/                  # Example using module_v2 (after upgrade)
+    ├── main.tf
+    ├── variables.tf
+    └── versions.tf
+```
+
+## Migration Guide for Module Maintainers
+
+### Step 1: Update the Cluster Resource
+
+Add `use_effective_fields = true` and remove the `lifecycle` block:
+
+**Before (module_v1):**
+```terraform
+resource "mongodbatlas_advanced_cluster" "this" {
+  project_id        = mongodbatlas_project.this.id
+  name              = var.cluster_name
+  cluster_type      = var.cluster_type
+  replication_specs = var.replication_specs
+  tags              = var.tags
+
+  # When auto-scaling is enabled (either compute or disk), Atlas may adjust
+  # all three fields (instance_size, disk_size_gb, disk_iops) regardless of
+  # which auto-scaling type is enabled. Must list all combinations of:
+  # - replication_specs and region_configs (based on cluster topology)
+  # - All 3 attributes for each node type used:
+  #   * electable_specs (required - always include)
+  #   * analytics_specs (optional - only if using analytics nodes)
+  #   * read_only_specs (optional - only if using read-only nodes)
+  lifecycle {
+    ignore_changes = [
+      replication_specs[0].region_configs[0].electable_specs.instance_size,
+      replication_specs[0].region_configs[0].electable_specs.disk_size_gb,
+      replication_specs[0].region_configs[0].electable_specs.disk_iops,
+      # ... and many more fields for all node types, regions, and shards
+    ]
+  }
+}
+```
+
+**After (module_v2):**
+```terraform
+resource "mongodbatlas_advanced_cluster" "this" {
+  project_id           = mongodbatlas_project.this.id
+  name                 = var.cluster_name
+  cluster_type         = var.cluster_type
+  use_effective_fields = true  # Replaces entire lifecycle block
+  replication_specs    = var.replication_specs
+  tags                 = var.tags
+}
+```
+
+### Step 2: Add Data Source for Effective Specs
+
+```terraform
+data "mongodbatlas_advanced_cluster" "this" {
+  project_id           = mongodbatlas_advanced_cluster.this.project_id
+  name                 = mongodbatlas_advanced_cluster.this.name
+  use_effective_fields = true
+  depends_on           = [mongodbatlas_advanced_cluster.this]
+}
+```
+
+### Step 3: Update Module Outputs
+
+Add outputs to expose effective specifications:
```

Comment:
Should we be explicit that if any outputs previously leveraged resource `replication_specs` and want to preserve the same resulting values, they now have to replaces the source of that output with the data source?

If they easily want to preserve all existing outputs they could even leverage a separate data source which does not use the `use_effective_fields`.